core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -109,14 +103,17 @@ final class KafkaMetadataLog private (
   }
 
   private def handleAndConvertLogAppendInfo(appendInfo: internals.log.LogAppendInfo): LogAppendInfo = {
-    if (appendInfo.firstOffset != JUnifiedLog.UNKNOWN_OFFSET)
+    if (appendInfo.firstOffset != UnifiedLog.UNKNOWN_OFFSET)
       new LogAppendInfo(appendInfo.firstOffset, appendInfo.lastOffset)
     else
       throw new KafkaException(s"Append failed unexpectedly")
   }
 
   override def lastFetchedEpoch: Int = {
-    log.latestEpoch.getOrElse {
+    val le = log.latestEpoch
+    if (le.isPresent)
+      le.get()
+    else {
```

Comment:
I was having issues with `orElse()` so resorted to the good old if/else.
TODO: Clean this up.